src/rules/at-mixin-no-risky-parent-selectors/__tests__/index.js

```diff
@@ -0,0 +1,194 @@
+"use strict";
+
+const { ruleName } = require("..");
+
+testRule({
+  ruleName,
+  config: [true],
+  customSyntax: "postcss-scss",
+
+  accept: [
+    {
+      code: `
+      .parent {
+        color: blue;
+    
+        & .b {
+          color: red;
+        }
+      }
+    `,
+      description: "Nested parent selector"
+    },
+    {
+      code: `
+      .parent {
+        color: blue;
+    
+        & .b, &.context {
+          color: red;
+        }
+      }
+    `,
+      description: "Nested parent selector in complex selector"
+    },
+    {
+      code: `
+      .bar {
+        & .parent {
+          color: blue;
+      
+          .context {
+            color: red;
+          }
+        }
+      }
+    `,
+      description: "Parent selector nested in another style rule"
+    },
+    {
+      code: `
+      @mixin foo {
+        &.context {
+          color: red;
+        }
+      }
+    `,
+      description: "Parent selector in mixin"
+    },
+    {
+      code: `
+      .bar {
+        .parent {
+          color: blue;
+      
+          & .context {
+            color: red;
+          }
+        }
+      }
+    `,
+      description: "Parent selector nested in more than one style rule"
+    },
+    {
+      code: `
+      .parent {
+        color: blue;
+    
+        .context & {
+          color: red;
+        }
+      }
+    `,
+      description: "Selector ending in parent selector"
+    },
+    {
+      code: `
+      .parent {
+        color: blue;
+    
+        .context & .b {
+          color: red;
+        }
+      }
+    `,
+      description: "Parent selector in the middle of complex selector"
+    },
+    {
+      code: `
+      .parent {
+        color: blue;
+    
+        & .b, .context & {
+          color: red;
+        }
+      }
+    `,
+      description: "Complex selector, one ending in parent selector"
+    },
+    {
+      code: `
+      @mixin foo {
+        .parent {
+          color: blue;
+      
+          & .context {
+            color: red;
+          }
+        }
+      }
+    `,
+      description: "Parent selector in mixin, nested, at the beginning"
+    },
+    {
+      code: `
+      @mixin foo {
+        .bar & {
+          color: red;
+        }
+      }
+    `,
+      description: "Parent selector in mixin, at the end"
+    },
+    {
+      code: `
+      @mixin foo {
+        .parent {
+          .bar, & {
+            color: red;
+          }
+        }
+      }
+    `,
+      description: "Complex selector in mixin, nested, no following class."
+    }
+  ],
+
+  reject: [
+    {
+      code: `
+      @mixin foo {
+        .bar {
+          color: blue;
+          .baz & {
+            color: red;
+          }
+        }
+      }
+    `,
```

Comment:
Please add `column`, `endColumn`, `endLine`, and `line` properties to all of these `reject` objects. That way we can verify that the error message is shown in a correct position.